packages/plugins/response-cache/src/plugin.ts

```diff
@@ -509,3 +548,51 @@ function calculateTtl(typeTtl: number, currentTtl: number | undefined): number {
 export const cacheControlDirective = /* GraphQL */ `
   directive @cacheControl(maxAge: Int) on FIELD_DEFINITION | OBJECT
 `;
+
+function mergePatch(
+  result: ExecutionResult,
+  patch: IncrementalDeferResult | IncrementalStreamResult,
+) {
+  // All errors and extensions are merged together in the final result
+  if (patch.errors) {
+    result.errors = [...(result.errors ?? []), ...patch.errors];
+  }
+
+  if (patch.extensions) {
+    result.extensions = { ...result.extensions, ...patch.extensions };
+  }
+
+  // We need to follow the path to the point where the patch should be applied
+  let target: any = result;
+  const path = ['data', ...(patch.path ?? [])];
+  for (let i = 0; i < path.length - 1; i++) {
+    target = target[path[i]];
+  }
+  const prop = path[path.length - 1];
+
+  const items = (patch as IncrementalStreamResult).items;
+  const data = (patch as IncrementalDeferResult).data;
```

Comment:
And I have the same Typescript problem here, where I can't discriminate the union type.